worker/src/sidechain_storage/storage.rs

```diff
@@ -0,0 +1,1062 @@
+		}
+		// remove shard from list
+		// STORED_SHARDS_KEY -> Vec<(Shard)>
+		self.shards.retain(|&x| x != *shard);
+		// add updated shards to Batch DB
+		SidechainDB::add_to_batch(&mut batch, STORED_SHARDS_KEY, &self.shards);
+		// Update DB
+		self.db.write(batch)
+	}
+
+	/// purges a shard and its block from the db storage
+	/// FIXME: Add delete functions?
+	pub fn prune_shard_from_block_number(
+		&mut self,
+		shard: &ShardIdentifierFor<SignedBlock>,
+		block_number: BlockNumber,
+	) -> Result<()> {
+		// Early return if shard is inexistent
+		if !self.shards.contains(shard) {
+			// check if DB is also empty
+			let db_shards = self.load_shards_from_db()?;
+			if !db_shards.contains(shard) {
+				warn!("Tried to prune already empty shard: {:?}", shard);
+				return Ok(())
+			}
+			warn!(
+				"Mismatch detected between inmemory shards and db shards, maybe try to reload DB? Guilty shard: {:?}", shard
+			);
+		}
+		// get last block of shard
+		let last_block = match self.last_blocks.get(shard) {
+			Some(last_block) => *last_block,
+			None => {
+				// Try to read from db:
+				self.load_last_block_from_db(shard)?
+					.ok_or_else(|| Error::LastBlockNotFound(format!("{:?}", *shard)))?
+			},
+		};
+		if last_block.number == block_number {
+			// given block number is last block of chain - purge whole shard
+			self.purge_shard(shard)
+		} else {
+			// iterate through chain and add all blocks to WriteBatch (delete cmd)
+			let mut batch = WriteBatch::default();
+			let mut current_block_number = block_number;
+			// Remove blocks from db until no block anymore
+			while let Some(block_hash) = self.get_block_hash(&shard, current_block_number)? {
+				self.delete_block(&mut batch, &block_hash, &current_block_number, shard);
+				current_block_number -= 1;
+			}
+			// Update DB
+			self.db.write(batch)
+		}
+	}
+
+	/// prunes all shards except for the newest blocks (according to blocknumber)
+	pub fn prune_shards(&mut self, blocks_to_keep: BlockNumber) {
+		for shard in self.shards().clone() {
+			// get last block:
+			if let Some(last_block) = self.last_block_of_shard(&shard) {
+				let threshold_block = last_block.number - blocks_to_keep;
+				if let Err(e) = self.prune_shard_from_block_number(&shard, threshold_block) {
+					error!("Could not purge shard {:?} due to {:?}", shard, e);
+				}
+			} else {
+				error!("Last block not found in shard {:?}", shard);
+			}
+		}
+	}
+
+	/// implementations of helper functions, not meant for pub use
+	/// gets the previous block of given shard and block number, if there is one
+	fn get_previous_block(
+		&self,
+		shard: &ShardIdentifierFor<SignedBlock>,
+		current_block_number: BlockNumber,
+	) -> Result<Option<LastSidechainBlock>> {
+		let prev_block_number = current_block_number - 1;
+		if let Some(block_hash) = self.get_block_hash(shard, prev_block_number)? {
+			Ok(Some(LastSidechainBlock { hash: block_hash, number: prev_block_number }))
+		} else {
+			Ok(None)
+		}
```

Comment:
```suggestion
		Ok(self.get_block_hash(shard, prev_block_number)?
		.map_or_default(|block_hash| 
			Some(LastSidechainBlock { hash: block_hash, number: prev_block_number })
	       ))

```

Reply:
Applied, but a little different (`map_or_default `not available for `Option`)